python/lsst/cp/pipe/cpFunctions.py

```diff
@@ -0,0 +1,516 @@
+# This file is part of cp_pipe.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If
+
+# Utility tasks/methods copied from pipe_drivers/constructCalibs.py
+
+import datetime
+import numpy as np
+import time
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+import lsst.afw.math as afwMath
+import lsst.afw.image as afwImage
+import lsst.daf.base as dafBase
+
+
+class BlessCalibration(pipeBase.Task):
+    """Create a way to bless existing calibration products.
+
+    The inputs are assumed to have been constructed via cp_pipe, and
+    already exist in the butler.
+
+    Parameters
+    ----------
+    kwargs : `dict`
+        Parameters to initialize the blessing process.  Keys:
+
+        ``"butler"``
+            Data butler to use.
+        ``"inputCollection"``
+            Data collection to pull calibrations from.
+        ``"outputCollection"``
+            Data collection to store final calibrations.
+    """
+    def __init__(self, **kwargs):
+        self.butler = kwargs['butler']
+        self.registry = self.butler.registry
+
+        self.inputCollection = kwargs['inputCollection']
+        self.outputCollection = kwargs['outputCollection']
+
+        self.calibrationLabel = ''
+        self.instrument = ''
+
+    def findInputs(self, datasetTypeName, inputTypeString=None):
+        """Find and prepare inputs for blessing.
+
+        Parameters
+        ----------
+        datasetTypeName : `str`
+            Dataset that will be blessed.
+        inputTypeString : `str`
+            Dataset name for the input datasets.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if no input datasets found or if the calibration
+            label exists and is not empty.
+        """
+        if inputTypeString is None:
+            inputDatasetTypeName = datasetTypeName + "Proposal"
```

Comment:
What is `inputDatasetTypeName` if `inputTypeString` is not `None`?

Reply:
This looks like a typo that wasn't detected by my testing.  Renaming inputTypeString to inputDatasetTypeName does what I want: allow searching for a default inputDatasetTypeName ("biasProposal" for example) when you're trying to bless the given datasetTypeName ("bias"), but let this be overridden if needed.